### PR TITLE
fix(08-wasm/test): register mockVm querier callbacks against TimestampAtHeightMsg to avoid recursive loops

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Register querier callbacks in mockVM against `TimestampAtHeightMsg` instead of `StatusMsg`.
This avoids recursive loops introduced by yours truly in https://github.com/cosmos/ibc-go/pull/5870, adding the client status check to the rpc handler of `VerifyMembership`.

If anyone would rather a different approach to fixing this issue, we could additional create an entirely separate test func for `success: verify membership query` and only use `TimestampAtHeightMsg` for that one, but it makes no difference to me.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/light-clients/08-wasm/types/querier_test.go [hunk 1]
```diff
@@ -286,7 +286,11 @@ func (suite *TypesTestSuite) TestStargateQuery() {
 
 			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), endpoint.ClientID)
 			clientState := endpoint.GetClientState()
-			clientState.Status(suite.chainA.GetContext(), clientStore, suite.chainA.App.AppCodec())
+
+			// NOTE: we register query callbacks against: types.TimestampAtHeightMsg{}
+			// in practise, this can against any client state msg, however registering against types.StatusMsg{} introduces recursive loops
```
[the remark this answers is not in the document]
Comment (colin-axner) [reply modules/light-clients/08-wasm/types/querier_test.go:291]: do we need to protect against this somehow? Or are avoiding recursive loops up to contract developers?

**Review by colin-axner — APPROVED** [review]
ACK, quick fix!

**Comment by colin-axner** [comment]
should `VerifyMembership` conditional queries only be allowed on 08-wasm `verify membership` contract api? Feels like we should disable in other scenarios? But if the contract specifies its own clientID, it would be recursive regardless?

**Comment by damiannolan** [comment]
> should `VerifyMembership` conditional queries only be allowed on 08-wasm `verify membership` contract api? Feels like we should disable in other scenarios? But if the contract specifies its own clientID, it would be recursive regardless?

Yeah, its down to the implementation of the contract right? and which one its calling. But specifying its own clientID seems like something that shouldn't ever need to happen - we can probably add a defensive check here. Happy to chat about this in eng call next week.

**Pull request merged**